core/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -168,6 +178,13 @@ private[spark] abstract class YarnSchedulerBackend(
         logInfo(s"ApplicationMaster registered as $am")
         amEndpoint = Option(am)
 
+      case ReregisterClusterManager(am) =>
```

Comment:
Is it robust enough to use `amEndpoint` to distinguish whether AM is registered again or not? Is it possible that this message may be duplicated but AM is only register once?